app/managers/apps_manager.test.ts

```diff
@@ -0,0 +1,289 @@
+
+        const mockDatabase = {
+            database: {},
+            operator: {},
+        };
+        DatabaseManager.getServerDatabaseAndOperator = jest.fn();
+        (DatabaseManager.getServerDatabaseAndOperator as jest.Mock).mockReturnValue(mockDatabase);
+
+        const mockClient = {
+            getAppsBindings: jest.fn().mockResolvedValue(mockBindings),
+        };
+        NetworkManager.getClient = jest.fn();
+        (NetworkManager.getClient as jest.Mock).mockReturnValue(mockClient);
+
+        AppsManager = new AppsManagerClass();
+    });
+
+    it('initializes with empty bindings', () => {
+        const bindings = AppsManager.getBindings(mockServerUrl);
+        expect(bindings).toEqual([]);
+    });
+
+    describe('apps enabled state', () => {
+        beforeEach(() => {
+            const mockConfig = {
+                FeatureFlagAppsEnabled: 'true',
+            } as ClientConfig;
+            jest.mocked(getConfig).mockResolvedValue(mockConfig);
+        });
+
+        it('should return true when apps are enabled', async () => {
+            const enabled = await AppsManager.isAppsEnabled(mockServerUrl);
+            expect(enabled).toBe(true);
+        });
+
+        it('should return false when database error occurs', async () => {
+            (DatabaseManager.getServerDatabaseAndOperator as jest.Mock).mockImplementation(() => {
+                throw new Error('Database error');
+            });
+
+            const enabled = await AppsManager.isAppsEnabled(mockServerUrl);
+            expect(enabled).toBe(false);
+        });
+
+        it('should observe apps enabled state', async () => {
+            const mockObservable = of$(false);
+            jest.spyOn(AppsManager, 'observeIsAppsEnabled').mockReturnValue(mockObservable);
+
+            const result = await new Promise((resolve) => {
+                AppsManager.observeIsAppsEnabled(mockServerUrl).subscribe((enabled) => {
+                    resolve(enabled);
+                });
+            });
+
+            expect(result).toBe(false);
+        });
+
+        it('should handle database error in observeIsAppsEnabled', async () => {
+            (DatabaseManager.getServerDatabaseAndOperator as jest.Mock).mockImplementation(() => {
+                throw new Error('Database error');
+            });
+
+            const result = await new Promise((resolve) => {
+                AppsManager.observeIsAppsEnabled(mockServerUrl).subscribe((enabled) => {
+                    resolve(enabled);
+                });
+            });
+
+            expect(result).toBe(false);
+        });
+    });
+
+    describe('bindings management', () => {
+        it('should fetch bindings successfully', async () => {
+            await AppsManager.fetchBindings(mockServerUrl, mockChannelId);
+            expect(NetworkManager.getClient).toHaveBeenCalledWith(mockServerUrl);
+            expect(NetworkManager.getClient(mockServerUrl).getAppsBindings).toHaveBeenCalled();
```

Comment:
A few of these tests don't really test much since the functions aren't structured in a way conducive to testing. However, this code will be removed in the near future as the Apps Framework has been deprecated so this is fine for coverage purposes for now.